pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -0,0 +1,268 @@
+        cpu_offload=None,
+        backward_prefetch=None,
+    ) -> None:
+        """Plugin for Fully Sharded Data Parallel provided by Pytorch.Distributed.
+
+        Full Sharded Training shards the entire model across all available GPUs, allowing you to scale model
+        size, whilst using efficient communication to reduce overhead. In practice, this means we can remain
+        at parity with PyTorch DDP, whilst scaling our model sizes dramatically. The technique is similar
+        to ZeRO-Stage 3.
+        `For more information: https://pytorch.org/blog/introducing-pytorch-fully-sharded-data-parallel-api/`.
+
+        .. warning:: ``DDPFullyShardedNativeStrategy`` is in beta and subject to change. The interface can
+        bring breaking changes and new features with the next release of Pytorch.
+
+        Defaults have been set and options have been exposed, but may require configuration
+        based on your level of memory/speed efficiency. We suggest having a look at this tutorial for
+        more information.
+        `https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html`
+
+        Arguments:
+            cpu_offload (Optional [CPUOffload]):
+                CPU offloading config. Currently, only parameter and gradient CPU
+                offload is supported. It can be enabled via passing in
+                ``cpu_offload=CPUOffload(offload_params=True)``. Note that this
+                currently implicitly enables gradient offloading to CPU in order for
+                params and grads to be on same device to work with optimizer. This
+                API is subject to change. Default is ``None`` in which case there
+                will be no offloading.
+            backward_prefetch: (Optional[BackwardPrefetch]):
+                This is an experimental feature that is subject to change in the
+                the near future. It allows users to enable two different backward_prefetch
+                algorithms to help backward communication and computation overlapping.
+                Pros and cons of each algorithm is explained in the class ``BackwardPrefetch``.
+        """
+        if not _TORCH_GREATER_EQUAL_1_11:
+            raise MisconfigurationException(
+                "`DDPFullyShardedNativeStrategy` is supported from pytorch v1.11.0 onwards."
+            )
+
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            cluster_environment=cluster_environment,
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+        self._process_group = None
+        self.num_processes = len(self.parallel_devices) if self.parallel_devices is not None else 0
+        self._process_group_backend: Optional[str] = process_group_backend
+        self.cpu_offload: Optional[CPUOffload] = cpu_offload
+        self.backward_prefetch: Optional[BackwardPrefetch] = backward_prefetch
```

Comment:
are these attributes expected to be mutable, or read-only once the strategy is constructed?

Reply:
We can pass in None for these attributes and they can take on default values that FSDP provides.